internal/server/server.go

```diff
@@ -1303,7 +1303,10 @@ func New(ctx context.Context, config *lepconfig.Config, endpoint string, cliUID
 		}
 	}()
 
-	if err = login.Gossip(endpoint, uid, config.Address); err != nil {
+	ghler.componentNamesMu.RLock()
+	componentNames := ghler.componentNames
+	ghler.componentNamesMu.RUnlock()
```

Comment:
@gyuho ptal, not sure if this is the best way to get most up to date component names..

Reply:
Correct. This is the latest set of components that are registers:

```go
				ghler.componentNamesMu.Lock()
				ghler.componentNames = newComponentNames
				ghler.componentNamesMu.Unlock()
			}
```